README.md
# Wipi-child-theme-on-steroids Version 1.0.5
Child theme on Stroids for WiPi theme! comes built-in with many features like: Hidden WP version, Disabled unnecessary functions,  login hints in wp-login are disabled, Custom Admin footer text and many more features...

#Installation
1) Download the files in .zip

2) Upload the child theme to your WP website under Appearance>Themes>Add New>Upload Theme like any theme.

3) Activate the theme and here you go - you have WiPi child theme on Steroids ready to go!

# Included functions

1. WordPress version hidden. - Security

2. Emoji icons feature disabled. - Performance 

3. Option to edit themes and plugins files from Admin panel disabled. - Security

4. Login hints in wp-login are disabled. - Security

5. Option to change Admin panel footer text added. - Appearance

6. Option to change wp-login logo to custom logo. - Appearance

7. Option to change wp-login logo url to custom url. - Appearance

8. Option to change wp-login logo title to custom title. - Appearance

9. JS loaded in footer instead in header. - Performance 

10. Google CDN for loading Jquery. - Performance 

# Links
WiPi web - https://www.wipi.co.il/

WiPi child theme on Steroids Author web - https://www.davidrevivo.co.il/

WiPi child theme on steroids Author post (Hebrew) - https://www.davidrevivo.co.il/wipi-child-theme-%D7%AA%D7%91%D7%A0%D7%99%D7%AA-%D7%91%D7%AA/

# Thanks
Shlomi Porush





changelog.txt
26.06.16 - Version 1.0.0: WiPi Child theme created
26.06.16 - Version 1.0.1:  rtl.css and screenshot.png added for WiPi Child theme
26.06.16 - Version 1.0.2: 6 New features added to functions.php
26.06.16 - Version 1.0.3: 2 New features added to functions.php
27.06.16 - Version 1.0.4: 3 New features added to functions.php
30.07.16 - Version 1.0.5: "Remove static Query strings" function deleted (cause for some bugs in WiPi parent theme V1.0.1)


functions.php
<?php
/**
 * Functions.php file for WiPi child theme on Steroids.
 *
 * @package   wipi-child-theme-on-steroids
 * @version   1.0.6
 * @link      http://www.davidrevivo.co.il/
 * @author    David Israel Revivo
 * @copyright Copyright (c) 2016, David Israel Revivo
 * @license   GPL-2.0+
 */

 <?php
// Enqueue css files from parent theme
add_action( 'wp_enqueue_scripts', 'my_theme_enqueue_styles' );
function my_theme_enqueue_styles() {
    wp_enqueue_style( 'parent-style', get_template_directory_uri() . '/style.css' );

}

// 1. Hide WordPress version
remove_action('wp_head', 'wp_generator');

// 2. Disable emoji icons
function disable_wp_emojicons() {

  // all actions related to emojis
  remove_action( 'admin_print_styles', 'print_emoji_styles' );

  remove_action( 'wp_head', 'print_emoji_detection_script', 7 );

  remove_action( 'admin_print_scripts', 'print_emoji_detection_script' );

  remove_action( 'wp_print_styles', 'print_emoji_styles' );

  remove_filter( 'wp_mail', 'wp_staticize_emoji_for_email' );

  remove_filter( 'the_content_feed', 'wp_staticize_emoji' );

  remove_filter( 'comment_text_rss', 'wp_staticize_emoji' );

  // filter to remove TinyMCE emojis
  add_filter( 'tiny_mce_plugins', 'disable_emojicons_tinymce' );

}

add_action( 'init', 'disable_wp_emojicons' );


// 3. Disable the themes and plugins text editor in Admin panel
define('DISALLOW_FILE_EDIT', true);

// 4. single login error message in wp-login
function wpfme_login_obscure(){ return '<strong>אופס!</strong> נראה שהפרטים שגויים!';}
add_filter( 'login_errors', 'wpfme_login_obscure' );

// 5. Change Admin footer text
function remove_footer_admin () { return '<a href="https://www.davidrevivo.co.il/wipi-child-theme-%D7%AA%D7%91%D7%A0%D7%99%D7%AA-%D7%91%D7%AA/" alt="WiPi Child theme on steroids to download">WiPi Child Theme on Steroids</a>'; } add_filter('admin_footer_text', 'remove_footer_admin');

// 6. Change wp-login Logo
function my_custom_login_logo() { echo '<style type="text/css"> h1 a { background-image: url(http://your-domain.com/image-path.png) !important; }
 </style>'; } add_action('login_head', 'my_custom_login_logo');

// 7. Change wp-login logo url
function custom_login_url() {
    return home_url();
}
add_filter( 'login_headerurl', 'custom_login_url' );

// 8. Change wp-login logo title
function custom_login_title() {
    return get_option( 'blogname' );
}
add_filter( 'login_headertitle', 'custom_login_title' );

// 9. Load js in footer
function remove_head_scripts() {
   remove_action('wp_head', 'wp_print_scripts');
   remove_action('wp_head', 'wp_print_head_scripts', 9);
   remove_action('wp_head', 'wp_enqueue_scripts', 1);
 
   add_action('wp_footer', 'wp_print_scripts', 5);
   add_action('wp_footer', 'wp_enqueue_scripts', 5);
   add_action('wp_footer', 'wp_print_head_scripts', 5);
}
add_action( 'wp_enqueue_scripts', 'remove_head_scripts' );

// 10. Use Google CDN to load Jquery
function savvy_remove_jquery_migrate_use_google_hosted_jquery() {
    if (!is_admin()) {
        wp_deregister_script('jquery');
        wp_register_script('jquery', 'https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js', false, null);
        wp_enqueue_script('jquery');
    }
}
add_action('init', 'savvy_remove_jquery_migrate_use_google_hosted_jquery');

?>


lib/features.php
<?php
// Security functions:
// Hide WordPress version
$hide_wp_version = get_field('hide_wp_version', 'option');
if ($hide_wp_version == "yes") {
  remove_action('wp_head', 'wp_generator');
}
// Disable the themes and plugins text editor in Admin panel
$disable_edit_in_control_panel = get_field('disable_edit_in_control_panel', 'option');
if ($disable_edit_in_control_panel == "yes") {
  define('DISALLOW_FILE_EDIT', true);
}
// single login error message in wp-login
$hide_wp_login_errors = get_field('hide_wp_login_errors', 'option');
if ($hide_wp_login_errors == "yes") {
  function wpfme_login_obscure() {
    $single_login_error_message = get_field('single_login_error_message', 'option');
    return $single_login_error_message;
  }
  add_filter( 'login_errors', 'wpfme_login_obscure' );
}
// Appereance functions:
// Change Admin footer text
function remove_footer_admin () {
  $change_wp_admin_footer_text = get_field('change_wp_admin_footer_text', 'option');
  return $change_wp_admin_footer_text;
}
add_filter('admin_footer_text', 'remove_footer_admin');

// Change wp-login Logo
function my_custom_login_logo() {
  $change_wp_login_logo = get_field('change_wp_login_logo', 'option');
  echo '<style type="text/css"> h1 a { background-image: url("'. $change_wp_login_logo .'") !important; }
  </style>';
}
add_action('login_head', 'my_custom_login_logo');

// Change wp-login logo url
function custom_login_url() {
  $change_wp_login_logo_url = get_field('change_wp_login_logo_url', 'option');
  return $change_wp_login_logo_url;
}
add_filter( 'login_headerurl', 'custom_login_url' );

// Change wp-login logo title
function custom_login_title() {
  $change_wp_login_title = get_field('change_wp_login_title', 'option');
  return $change_wp_login_title;
}
add_filter( 'login_headertitle', 'custom_login_title' );

// Performance functions:
// Load js in footer
$load_js_in_footer = get_field('load_js_in_footer', 'option');
if ($load_js_in_footer == "yes") {
  function remove_head_scripts() {
    remove_action('wp_head', 'wp_print_scripts');
    remove_action('wp_head', 'wp_print_head_scripts', 9);
    remove_action('wp_head', 'wp_enqueue_scripts', 1);

    add_action('wp_footer', 'wp_print_scripts', 5);
    add_action('wp_footer', 'wp_enqueue_scripts', 5);
    add_action('wp_footer', 'wp_print_head_scripts', 5);
  }
  add_action( 'wp_enqueue_scripts', 'remove_head_scripts' );
}
// Use Google CDN to load Jquery
$load_jquery_google_cdn = get_field('load_jquery_google_cdn', 'option');
if ($load_jquery_google_cdn == "yes") {
  function google_jquery_cdn() {
    if (!is_admin()) {
      wp_deregister_script('jquery');
      wp_register_script('jquery', 'https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js', false, null);
      wp_enqueue_script('jquery');
    }
  }
  add_action('init', 'google_jquery_cdn');
}
// Disable emoji icons
$disable_emoji = get_field('disable_emoji', 'option');
if ($disable_emoji == "yes") {
  function disable_wp_emojicons() {
    // all actions related to emojis
    remove_action( 'admin_print_styles', 'print_emoji_styles' );
    remove_action( 'wp_head', 'print_emoji_detection_script', 7 );
    remove_action( 'admin_print_scripts', 'print_emoji_detection_script' );
    remove_action( 'wp_print_styles', 'print_emoji_styles' );
    remove_filter( 'wp_mail', 'wp_staticize_emoji_for_email' );
    remove_filter( 'the_content_feed', 'wp_staticize_emoji' );
    remove_filter( 'comment_text_rss', 'wp_staticize_emoji' );
    // filter to remove TinyMCE emojis
    /* Warning: If you have TinyMCE plugin I
    highly recommended to remove this function because this function cause to
    some bugs like Color text button doesnt appear, etc. */
    add_filter( 'tiny_mce_plugins', 'disable_emojicons_tinymce' );
  }
  add_action( 'init', 'disable_wp_emojicons' );
}

// Remove Query strings
$remove_query_strings = get_field('remove_query_strings', 'option');
if ($remove_query_strings == "yes") {
  function remove_script_version( $src ){
    $parts = explode( '?ver', $src );
      return $parts[0];
  }
  add_filter( 'script_loader_src', 'remove_script_version', 15, 1 );
  add_filter( 'style_loader_src', 'remove_script_version', 15, 1 );
}
 ?>


lib/cfg-acf.php
<?php
// Import ACF Pro Custom fields groups
if( function_exists('acf_add_local_field_group') ):

acf_add_local_field_group(array (
  'key' => 'group_57757b161ed9e',
  'title' => 'אבטחה',
  'fields' => array (
    array (
      'key' => 'field_57757b258d0a9',
      'label' => 'הסתרת גרסת וורדפרס',
      'name' => 'hide_wp_version',
      'type' => 'select',
      'instructions' => 'האם תרצו להסתיר את גרסת הוורדפרס של האתר?
<br>
- אם אתם משתמשים בתוסף WordFence אין טעם להפעיל פונקציה זו כיוון שהיא מוגדרת כברירת מחדל.',
      'required' => 0,
      'conditional_logic' => 0,
      'wrapper' => array (
        'width' => 50,
        'class' => '',
        'id' => '',
      ),
      'choices' => array (
        'yes' => 'כן',
        'no' => 'לא',
      ),
      'default_value' => array (
        0 => 'yes',
      ),
      'allow_null' => 0,
      'multiple' => 0,
      'ui' => 1,
      'ajax' => 1,
      'placeholder' => '',
      'disabled' => 0,
      'readonly' => 0,
    ),
    array (
      'key' => 'field_57757c9477bb5',
      'label' => 'ביטול עריכת קבצי תבניות/ תוספים דרך לוח הבקרה',
      'name' => 'disable_edit_in_control_panel',
      'type' => 'select',
      'instructions' => 'האם תרצו לבטל את האפשרות לערוך את קבצי התבניות והתוספים דרך לוח הבקרה? (נמצא תחת עיצוב>ערוך)
',
      'required' => 0,
      'conditional_logic' => 0,
      'wrapper' => array (
        'width' => 50,
        'class' => '',
        'id' => '',
      ),
      'choices' => array (
        'yes' => 'כן',
        'no' => 'לא',
      ),
      'default_value' => array (
        0 => 'כן',
      ),
      'allow_null' => 0,
      'multiple' => 0,
      'ui' => 1,
      'ajax' => 1,
      'placeholder' => '',
      'disabled' => 0,
      'readonly' => 0,
    ),
    array (
      'key' => 'field_5775839b72ebc',
      'label' => 'הסתרת שגיאת התחברות',
      'name' => 'hide_wp_login_errors',
      'type' => 'select',
      'instructions' => 'האם תרצו להסתיר את שגיאות ההתחברות ובמקום זאת להציג למשתמשים שגיאת התחברות אחידה?',
      'required' => 0,
      'conditional_logic' => 0,
      'wrapper' => array (
        'width' => 50,
        'class' => '',
        'id' => '',
      ),
      'choices' => array (
        'yes' => 'כן',
        'no' => 'לא',
      ),
      'default_value' => array (
        0 => 'כן',
      ),
      'allow_null' => 0,
      'multiple' => 0,
      'ui' => 1,
      'ajax' => 1,
      'placeholder' => '',
      'disabled' => 0,
      'readonly' => 0,
    ),
    array (
      'key' => 'field_577691f97dd67',
      'label' => 'נוסח שגיאת התחברות אחידה',
      'name' => 'single_login_error_message',
      'type' => 'text',
      'instructions' => 'במידה ובחרתם להפעיל את הפונקציה "הסתרת שגיאות התחברות" עליכם לנסח את שגיאת ההתחברות שתוצג למשתמשים שלא יצליחו להתחבר.',
      'required' => 0,
      'conditional_logic' => array (
        array (
          array (
            'field' => 'field_5775839b72ebc',
            'operator' => '==',
            'value' => 'yes',
          ),
        ),
      ),
      'wrapper' => array (
        'width' => 50,
        'class' => '',
        'id' => '',
      ),
      'default_value' => 'אופס! נראה שהפרטים שגויים.',
      'placeholder' => '',
      'prepend' => '',
      'append' => '',
      'maxlength' => '',
      'readonly' => 0,
      'disabled' => 0,
    ),
  ),
  'location' => array (
    array (
      array (
        'param' => 'options_page',
        'operator' => '==',
        'value' => 'acf-options-theme-features',
      ),
    ),
  ),
  'menu_order' => 0,
  'position' => 'normal',
  'style' => 'default',
  'label_placement' => 'top',
  'instruction_placement' => 'label',
  'hide_on_screen' => '',
  'active' => 1,
  'description' => '',
));

acf_add_local_field_group(array (
  'key' => 'group_57758406323f4',
  'title' => 'ביצועים',
  'fields' => array (
    array (
      'key' => 'field_57758409497d1',
      'label' => 'טעינת קבצי JS בפוטר',
      'name' => 'load_js_in_footer',
      'type' => 'select',
      'instructions' => 'האם תרצו לטעון את קבצי ה-JS בפוטר במקום בהידר?',
      'required' => 0,
      'conditional_logic' => 0,
      'wrapper' => array (
        'width' => 50,
        'class' => '',
        'id' => '',
      ),
      'choices' => array (
        'yes' => 'כן',
        'no' => 'לא',
      ),
      'default_value' => array (
        0 => 'כן',
      ),
      'allow_null' => 0,
      'multiple' => 0,
      'ui' => 1,
      'ajax' => 1,
      'placeholder' => '',
      'disabled' => 0,
      'readonly' => 0,
    ),
    array (
      'key' => 'field_5775842c497d2',
      'label' => 'טעינת Jquery מ-Google CDN',
      'name' => 'load_jquery_google_cdn',
      'type' => 'select',
      'instructions' => 'האם תרצו לטעון את ה-Jquery דרך ה-CDN של גוגל?',
      'required' => 0,
      'conditional_logic' => 0,
      'wrapper' => array (
        'width' => 50,
        'class' => '',
        'id' => '',
      ),
      'choices' => array (
        'yes' => 'כן',
        'no' => 'לא',
      ),
      'default_value' => array (
        0 => 'כן',
      ),
      'allow_null' => 0,
      'multiple' => 0,
      'ui' => 1,
      'ajax' => 1,
      'placeholder' => '',
      'disabled' => 0,
      'readonly' => 0,
    ),
    array (
      'key' => 'field_57758462497d3',
      'label' => 'ביטול פיצ\'ר הסמיילים (Emoji)',
      'name' => 'disable_emoji',
      'type' => 'select',
      'instructions' => 'האם תרצו להסיר את הסמיילים באתר שלכם?
<br>
- אם אתם משתמשים בתוסף TinyMCE כדאי לא להפעיל פונקציה זו.',
      'required' => 0,
      'conditional_logic' => 0,
      'wrapper' => array (
        'width' => 50,
        'class' => '',
        'id' => '',
      ),
      'choices' => array (
        'yes' => 'כן',
        'no' => 'לא',
      ),
      'default_value' => array (
        0 => 'כן',
      ),
      'allow_null' => 0,
      'multiple' => 0,
      'ui' => 1,
      'ajax' => 1,
      'placeholder' => '',
      'disabled' => 0,
      'readonly' => 0,
    ),
    array (
      'key' => 'field_57782faae5184',
      'label' => 'מחיקת Query Strings',
      'name' => 'remove_query_strings',
      'type' => 'select',
      'instructions' => 'האם תרצו להסיר Query strings מקבצים סטטיים?
',
      'required' => 0,
      'conditional_logic' => 0,
      'wrapper' => array (
        'width' => 50,
        'class' => '',
        'id' => '',
      ),
      'choices' => array (
        'yes' => 'כן',
        'no' => 'לא',
      ),
      'default_value' => array (
        0 => 'לא',
      ),
      'allow_null' => 0,
      'multiple' => 0,
      'ui' => 1,
      'ajax' => 1,
      'placeholder' => '',
      'disabled' => 0,
      'readonly' => 0,
    ),
  ),
  'location' => array (
    array (
      array (
        'param' => 'options_page',
        'operator' => '==',
        'value' => 'acf-options-theme-features',
      ),
    ),
  ),
  'menu_order' => 0,
  'position' => 'normal',
  'style' => 'default',
  'label_placement' => 'top',
  'instruction_placement' => 'label',
  'hide_on_screen' => '',
  'active' => 1,
  'description' => '',
));

acf_add_local_field_group(array (
  'key' => 'group_57768062bc422',
  'title' => 'נראות',
  'fields' => array (
    array (
      'key' => 'field_5776807c91312',
      'label' => 'שינוי טקסט הפוטר בלוח הבקרה',
      'name' => 'change_wp_admin_footer_text',
      'type' => 'text',
      'instructions' => 'האם תרצו לשנות את הטקסט בפוטר בלוח הבקרה?
<br>
ברירת המחדל היא: "WiPi Child Theme on Steroids"',
      'required' => 0,
      'conditional_logic' => 0,
      'wrapper' => array (
        'width' => 50,
        'class' => '',
        'id' => '',
      ),
      'default_value' => '<a href="https://www.davidrevivo.co.il/wipi-child-theme-%D7%AA%D7%91%D7%A0%D7%99%D7%AA-%D7%91%D7%AA/" alt="WiPi Child theme on steroids to download">WiPi Child Theme on Steroids</a>',
      'placeholder' => '',
      'prepend' => '',
      'append' => '',
      'maxlength' => '',
      'readonly' => 0,
      'disabled' => 0,
    ),
    array (
      'key' => 'field_5776811191313',
      'label' => 'לוגו מותאם אישית בטופס ההתחברות',
      'name' => 'change_wp_login_logo',
      'type' => 'image',
      'instructions' => 'על הלוגו להיות בגודל 80px*80px',
      'required' => 0,
      'conditional_logic' => 0,
      'wrapper' => array (
        'width' => 50,
        'class' => '',
        'id' => '',
      ),
      'return_format' => 'url',
      'preview_size' => 'full',
      'library' => 'all',
      'min_width' => 80,
      'min_height' => 80,
      'min_size' => '',
      'max_width' => 80,
      'max_height' => 80,
      'max_size' => '',
      'mime_types' => '',
    ),
    array (
      'key' => 'field_577681a091314',
      'label' => 'שינוי הurl של הלוגו בטופס ההתחברות',
      'name' => 'change_wp_login_logo_url',
      'type' => 'url',
      'instructions' => 'ברירת המחדל היא wordpress.org',
      'required' => 0,
      'conditional_logic' => 0,
      'wrapper' => array (
        'width' => 50,
        'class' => '',
        'id' => '',
      ),
      'default_value' => 'http://www.wordpress.org/',
      'placeholder' => '',
    ),
    array (
      'key' => 'field_577681f291315',
      'label' => 'שינוי כותרת הלוגו בטופס ההתחברות',
      'name' => 'change_wp_login_title',
      'type' => 'text',
      'instructions' => 'ברירת המחדל היא "פועל על וורדפרס"',
      'required' => 0,
      'conditional_logic' => 0,
      'wrapper' => array (
        'width' => 50,
        'class' => '',
        'id' => '',
      ),
      'default_value' => 'פועל על וורדפרס',
      'placeholder' => '',
      'prepend' => '',
      'append' => '',
      'maxlength' => '',
      'readonly' => 0,
      'disabled' => 0,
    ),
  ),
  'location' => array (
    array (
      array (
        'param' => 'options_page',
        'operator' => '==',
        'value' => 'acf-options-theme-features',
      ),
    ),
  ),
  'menu_order' => 0,
  'position' => 'normal',
  'style' => 'default',
  'label_placement' => 'top',
  'instruction_placement' => 'label',
  'hide_on_screen' => '',
  'active' => 1,
  'description' => '',
));

endif;

?>
